fedimint-client/src/module/gen.rs

```diff
@@ -98,10 +97,10 @@ where
         notifier: Notifier<DynGlobalClientContext>,
         api: DynGlobalApi,
     ) -> anyhow::Result<DynClientModule> {
-        let typed_cfg = cfg.cast::<T::Config>()?;
+        let typed_cfg: &<<T as fedimint_core::module::ExtendsCommonModuleGen>::Common as CommonModuleGen>::ClientConfig = cfg.cast()?;
```

Comment:
If you ever have a better idea how to deal with common/client/server module gens and their extensibility please lmk, I know I built it but it just looks ugly.